lighthouse-core/gather/driver.js

```diff
@@ -122,6 +122,28 @@ class Driver {
     };
   }
 
+  /**
+   * Reject if the gathering terminates due to the CDP target crashing, etc
+   * @return {Promise<void>}
+   */
+  async getGatherTerminatedPromise() {
+    /** @param {number} ms */
+    const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+    return new Promise((_, reject) => {
+      this.on('Inspector.targetCrashed', async _ => {
+        await wait(1000);
+        reject(new LHError(LHError.errors.TARGET_CRASHED));
+      });
+      // In case of crash, detached fires after targetCrashed, so we'll exit with the crash code
+      // Detachment happens (in non-crash cases) when the browser tab is closed or unexpected connection failure.
+      this.on('Inspector.detached', async _ => {
+        await wait(1000);
```

Comment:
Is the second because there is time between proper driver disposal and when GatherRunner.run resolves?

ie https://github.com/GoogleChrome/lighthouse/blob/de1fd107c42e12be977ccbc4c19c2ea7e1da2a2f/lighthouse-core/gather/gather-runner.js#L539-L540 just finalizeArtifacts runs between dispose and resolve, and if you look at that function it is not doing anything heavy. so a second will certainly cover that.